framework/src/systems/NonlinearSystemBase.C

```diff
@@ -167,6 +167,9 @@ NonlinearSystemBase::NonlinearSystemBase(FEProblemBase & fe_problem,
   _Re_tag = _fe_problem.addVectorTag("RESIDUAL");
 
   _u_dot = &addVector("u_dot", true, GHOSTED);
+  _u_dotdot = &addVector("u_dotdot", true, GHOSTED);
+  _u_dot_old = &addVector("u_dot_old", true, GHOSTED);
+  _u_dotdot_old = &addVector("u_dotdot_old", true, GHOSTED);
```

Comment:
So if I understand correctly, the `\ddot{u}` and old variants get added to `NonlinearSystemBase` regardless of whether they are actually being used by the application? Adding a handful of vectors is obviously OK for small problems but could be more troublesome for dof-heavy applications like neutronics, etc. @friedmud may have already commented on this, just wanted to mention it to be on the safe side...

Reply:
I assumed that Derek's comment only applied to computing the values at each quadrature point as is done in MooseVariableFE. I dint realize that storing these vectors would also use up a lot of memory. I will add if statements to add the udot, u dotdot vectors and the old values only when requested by the time integrator.

Reply:
I have made u_dot, u_dotdot, u_dot_old and u_dotdot_old to be optional vectors that are added only when requested by the time integrator. 

Earlier, the kernel, dirackernel, scalarkernel and nodalkernel base classes were requesting u_dot irrespective of whether it is steady state or transient problem. Now, I have moved u_dot calculation to TimeKernel, TimeNodalKernel and similar class for ScalarKernel. In a seperate PR, I will remove u_dot from the TimeKernel class and add it to the specific child class such as TimeDerivative that requires u_dot. I am not sure how many files will need to be modified for this change, so I will do it in separate PR later.